### DISCUSSION TITLE
Increase memory resource while retrying?

### DISCUSSION THREAD

**Discussion by aixi** [argoproj/argo-workflows#12482] [Q&A] [answered]
I found my argo workflow pod crash because of OOM，then the pod restart and OOM again. Can pod get more memory(it need dynamically change resource.requests.memory)

**Comment by shuangkun** [comment]
Yes, I think this is reasonable and achievable, we have done something before to capture the OOM event of the pod and increase the memory of the pod to 2 times when retrying. But it is not supported in the community yet.

**Reply by aixi** [reply]
Does Alibaba Cloud currently have any cloud products based on Argo workflow that also support GPU machines?
现在阿里云有基于argo workflow的云产品吗，同时支持gpu机器的？

**Reply by shuangkun** [reply]
Yes, fully managed Argo Workflows:
https://help.aliyun.com/zh/ack/distributed-cloud-container-platform-for-kubernetes/user-guide/overview-12?spm=a2c4g.11186623.0.0.4e10ccdfdRrufV

**Reply by shuangkun** [reply]
I can assist with the use of the product if needed.

**Comment by agilgur5** [comment, marked as answer]
Yes, we do have at least a couple of users doing something like this based [on Slack](https://cloud-native.slack.com/archives/C01QW9QSSSK/p1695730941009859?thread_ts=1695718281.588569&cid=C01QW9QSSSK) and issues. You can use `podSpecPatch` to implement this and can detect an `OOMKilled` event usually by its `137` exit code.

This will also be easier to do with the completion of #10362 and #10364

**Reply by aixi** [reply]
thanks, it's very helpful
